libraries/botbuilder/src/activityValidator.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Activity, ActivityTimestamps } from 'botbuilder-core';
+
+export function validateActivity(activity: Activity): Activity {
+    if (typeof activity !== 'object') { throw new Error(`validateActivity(): invalid request body.`); }
+    if (typeof activity.type !== 'string') { throw new Error(`validateActivity(): missing activity type.`); }
+    if (typeof activity.timestamp === 'string') {
+        (activity as ActivityTimestamps).rawTimestamp = activity.timestamp;
```

Comment:
It's a little weird to have a function called "validate" go ahead and change the thing its validating.
Validate _and fix_ might be more appropriate.

Reply:
Good point.  Changed to validateAndFixActivity